### ISSUE TITLE
[clang-tidy] "modernize-use-emplace" does not provide diagnostic for `std::optional`

### ISSUE THREAD

**Issue by denzor200** [llvm/llvm-project#133108]
```
std::vector<std::pair<std::string, int>> v;
v.push_back(std::pair<std::string, int>("first", 1)); // OK - clang tidy provides warnings for that

std::optional<std::pair<std::string, int>> o;
o = std::pair<std::string, int>{"first", 1};          // BAD - silence
```

The full snippet here: https://godbolt.org/z/nW9dKscqo

**Comment by vbvictor** [comment]
According to the docs, it should not warn on `operator=`.

> The check flags insertions to an STL-style container done by calling the push_back, push, or push_front methods with an explicitly-constructed temporary of the container element type.
